### PR TITLE
dev/core#1987: Fix theme configuration section on Display preference and improve `isFrontendPage`  function for Drupal CMS.

### PR DESCRIPTION
**Pull request by swastikpareek**
Overview
----------------------------------------
Drupal webform pages may have CiviCRM fields which internally initialize CiviCRM assets and was loading CiviCRM current theme assets. This is because CiviCRM doesn't have the logic for checking Drupal webform pages, whether they are FE pages or backend and since the logic is incomplete it loads the backend theme which because of [this logic](https://github.com/civicrm/civicrm-core/blob/master/Civi/Core/Themes.php#L72.).
Also, CiviCRM, to handle front end pages theme (public pages) and backend pages theme (CiviCRM pages) separately contains a configuration under `civicrm/admin/setting/preferences/display?reset=1`. Where it provides a configuration field for the backend and the front end theme separately. 
  This works for all CMS except Drupal. For Drupal, it only had the `Theme` option which configures only the backend theme. This PR enables both the options for Drupal CMS too and also makes sure the correct theme (front end theme) is loaded on Drupal webform pages which loads CiviCRM components.

Before
----------------------------------------
3ffa703e-9774-4224-abb0-c967b809dbef

Also, the CiviCRM was loading backend theme for the Drupal public web form pages which loads CiviCRM components.

After
----------------------------------------
3403e6d0-6a80-4e20-ac50-bd24190cc858

Also, the CiviCRM was loading the front theme for the Drupal public web form pages which loads CiviCRM components.

Technical Details
----------------------------------------
* `Display.tpl` was fixed so that both front end and backend theme configuration was provided to the end-user
* `isFrontEndPage` is updated so that it treats Drupal public web form pages as CiviCRM public pages and load front end theme there.

### CODE REVIEW

**Comment by mattwire** [comment]
@swastikpareek Thankyou. Can you submit two separate PRs (link to the same lab issue #1987) for the two issues. I agree that both issues should be fixed but they are different things.

File: CRM/Utils/System/DrupalBase.php [hunk 1]
```diff
@@ -660,7 +660,7 @@ public function isFrontEndPage() {
 
     // Get the menu for above URL.
     $item = CRM_Core_Menu::get($path);
-    return !empty($item['is_public']);
+    return (empty($item) || !empty($item['is_public']));
```
Comment (mattwire) [inline CRM/Utils/System/DrupalBase.php:663]: @swastikpareek Can you add comments here explaining why (use the comments you already made on dev/core#1987)

**Comment by seamuslee001** [comment]
Jenkins add to whitelist

**Comment by totten** [comment]
I think this is good idea.

The old conditional was put in there before the Drupal integration officially made a distinction (eg before 68e3bc34581df6cb175e8b5296950d790f4c357a). Now that it makes the distinction, it's important to show both settings.

**Comment by eileenmcnaughton** [comment]
I think we don't need a test as it is just removing logic & form layer

File: CRM/Utils/System/DrupalBase.php (hunk 1 above)
Comment (swastikpareek) [reply CRM/Utils/System/DrupalBase.php:663]: Sure will do.

**Comment by swastikpareek** [comment]
As per @mattwire, the PR is split into two different PR as they both are different issues
* For template - https://github.com/civicrm/civicrm-core/pull/18396
* For function - https://github.com/civicrm/civicrm-core/pull/18397

Please let me know if anything else needs to be done here.

Closing this PR. Thanks.

cc @totten @seamuslee001 @eileenmcnaughton

**Pull request closed without merging**